### ISSUE TITLE
ScrollController return maxScrollExtent as infinity when mouse scroll

### ISSUE THREAD

**Issue by Sasikumar3595** [flutter/flutter#75115]
I was created the flutter application with custom scroll view and load the scroll view data dynamically when the scroll view reaches it bottom. but the scroll view controller does return max scroll extent value as infinity on most of mouse scrolls on web and macOS(Issue replicated on beta and dev branch).

## Steps to Reproduce

1. Run the sample with below added code
2. scroll the scroll view with mouse(In Mac, using trackpad for scroll)
3. Issue raised on some fast scrolls.

```dart
class InfiniteScroll extends StatefulWidget {
  @override
  _InfiniteScrollState createState() => _InfiniteScrollState();
}

class _InfiniteScrollState extends State<InfiniteScroll> {
  final List<String> _data = <String>[];
  final ScrollController _controller =
      ScrollController(initialScrollOffset: 0.0, keepScrollOffset: true);
  final Key _key = UniqueKey();
  bool loading = false;

  @override
  void initState() {
    _loadData();
    _controller.addListener(() async{
      if(loading){
        return;
      }
      final bool isEnd =
          _controller.offset >= _controller.position.maxScrollExtent;
      if (isEnd) {
        print('Loaded' + (_data.length + 20).toString());
        setState(() {
        loading = true;
        });
        await Future.delayed(Duration(seconds: 2));
          _loadData();
      }
    });
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Infinite Scrolling'),
      ),
      body: Container(
        child: Stack(children: [
          CustomScrollView(
        physics: const AlwaysScrollableScrollPhysics(
            parent:
                ClampingScrollPhysics(parent: RangeMaintainingScrollPhysics())),
            controller: _controller,
            center: _key,
            slivers: <Widget>[
              SliverList(
                delegate:
                SliverChildBuilderDelegate((BuildContext context, int index) {
                  if (_data.length <= index) {
                    return null;
                  }
                  return Container(
                    height: 50,
                    child: Text(_data[index].toString()),
                  );
                }),
                key: _key,
              ),
            ],
          ),
          if(loading)
              Container(height: 50,alignment: Alignment.bottomCenter, child: CircularProgressIndicator())
        ],),
      )
    );
  }

  Future<List<String>> _loadData() async {
    final int length = _data.length;
    for (int i = length; i < length + 20; i++) {
      _data.add('Data item - $i');
    }

    if (loading) {
      setState(() {
        loading = false;
      });
    }

    return _data;
  }
}
```

**Expected results:** 
- Scroll controller max scroll extent return valid value

**Actual results:**
- Scroll controller max scroll extent return infinity value

**Replication Video:**

https://user-images.githubusercontent.com/44917578/106421581-fa222580-6482-11eb-9ae7-136676de2e11.mov

<details>
  <summary>Flutter doctor</summary>

```
[✓] Flutter (Channel dev, 1.26.0-17.1.pre, on Mac OS X 10.15.3 19D76 darwin-x64, locale en-US)                                                                                                                                                                                                    [!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[!] Xcode - develop for iOS and macOS
    ! Xcode 11.3.1 out of date (12.0.1 is recommended).
      Download the latest version or update via the Mac App Store.
    ✗ Xcode requires additional components to be installed in order to run.
      Launch Xcode and install additional required components when prompted or run:
        sudo xcodebuild -runFirstLaunch
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.52.0)
[✓] Connected device (3 available)
```

<details>

**Comment by darshankawar** [comment]
@Sasikumar3595 
I tried your code sample on latest master and scrolled down till `Data Item - 250` and saw below values being printed in the console:

```
Loaded40
Loaded60
Loaded80
Loaded100
Loaded120
Loaded140
Loaded160
Loaded180
Loaded200
Loaded220
Loaded240
Loaded260
```

Is this the expected behavior ? Your video only shows scrolling in UI, but not what gets printed in the console while doing so.

<details>
<summary>flutter doctor -v</summary>

```

[✓] Flutter (Channel master, 1.26.0-18.0.pre.90, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-IN)
    • Flutter version 1.26.0-18.0.pre.90 at
      /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 25134a16d1 (2 days ago), 2021-01-29 23:52:30 -0800
    • Engine revision ce14c8a841
    • Dart version 2.12.0 (build 2.12.0-276.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • SM A260G (mobile)            • 5200763ebcfa861f • android-arm    • Android
      8.1.0 (API 27)
    • AOSP on IA Emulator (mobile) • emulator-5554    • android-x86    • Android
      9 (API 28) (emulator)
    • macOS (desktop)              • macos            • darwin-x64     • Mac OS
      X 10.15.4 19E2269 darwin-x64
    • Chrome (web)                 • chrome           • web-javascript • Google
      Chrome 88.0.4324.96

! Doctor found issues in 1 category.

```

</details>

**Comment by Sasikumar3595** [comment]
@darshankawar,

Thanks for you response.

Yes, printing the loaded data count is expected. My actual issue is scroll controller max scroll extent value become infinity on sometimes scroll by trackpad. Above specified sample to add 20 items while scroll reaches it max scroll extent but some times items are not loaded because the max scroll extend value becomes infinity.

**Comment by darshankawar** [comment]
@Sasikumar3595 
Usually after what data item count do you start seeing the infinity issue ? I tried by scrolling till 400, but didn't see the issue.

**Comment by Sasikumar3595** [comment]
@darshankawar,

It was reproduced on some times fast scroll. In the above attached video, scroll to end(20) it loads another 20 items after that scroll reaches it end(39) it does not loads because the scroll controller max scroll extent value as infinity.

After that reaches the end(59) scroll also return infinity value.

**Note**:  Fast scrolling to reach it end reproduces the issue

**Comment by darshankawar** [comment]
@Sasikumar3595 
Tried to fast scroll, but still didn't see the issue. Also, what's the exact error you get in the console or in UI ?

<img width="558" alt="Screenshot 2021-02-02 at 12 43 48 PM" src="https://user-images.githubusercontent.com/67046386/106565244-83edf380-6554-11eb-98bf-c221afc5dc4e.png">

<details>
<summary>flutter doctor -v</summary>

```

[✓] Flutter (Channel master, 1.26.0-18.0.pre.121, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-IN)
    • Flutter version 1.26.0-18.0.pre.121 at
      /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 225e408d4f (3 hours ago), 2021-02-01 22:26:03 -0500
    • Engine revision e59457b818
    • Dart version 2.12.0 (build 2.12.0-279.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • SM A260G (mobile) • 5200763ebcfa861f • android-arm    • Android 8.1.0 (API
      27)
    • macOS (desktop)   • macos            • darwin-x64     • Mac OS X 10.15.4
      19E2269 darwin-x64
    • Chrome (web)      • chrome           • web-javascript • Google Chrome
      88.0.4324.96

! Doctor found issues in 1 category.

```

</details>

**Comment by Sasikumar3595** [comment]
@darshankawar,

The issue occurred on UI. I was created the application with custom scroll view and load(20) data to its data source when the scroll view reaches it end. I was identified the scroll end by scroll controller position's offset and scroll controller position's max scroll extent but some times scroll controller position's max scroll extent holds infinity value so i did not identify the scroll view reaches it end.

**Comment by darshankawar** [comment]
@Sasikumar3595 
I still don't understand 100% what exactly the issue is. Can you provide details around what should be the expected behavior against actual behavior when you scroll ? If you can provide screenshots that'll help to understand, it'll be good.

**Comment by Sasikumar3595** [comment]
@darshankawar,

I was implemented the lazy loading scroll view to load the data on runtime(on end of scroll). I was identified the end of scroll by scroll controller position's offset and scroll controller position's max scroll extent but some times scroll controller position's max scroll extent holds infinity value so i did not identify the scroll view reaches it end.

Eg., Initially loaded data from 0 -19 and load 20-39 when scroll view, scroll reach end of scroll(end of scroll after 19th item) and load 40-59 when scroll view, scroll reach end of scroll(end of scroll after 39th item).

In the above attached video, circle progress indicator loaded when i reach 19 item but it does not loaded when i reaches 39 item because its max scroll extent value is infinity.

**Comment by darshankawar** [comment]
@Sasikumar3595 
I tried again your code sample to understand the context and below is my observation:

1. Upon app launch, I see `Data item - 0 to 12`.
2. If I scroll up, I see `data item -19` and in console I see `Loaded40`.
3. Next scroll up, I see `data item - 39` and in console I see `Loaded60`.
4. Then `data item - 59` and `Loaded80`

Can you tell me in above context, if it wasn't expected ? or what should have been the behavior ?

**Comment by Sasikumar3595** [comment]
@darshankawar,

1. Initially loaded data item 0 -19
2. Scroll up, when you reached data item -19, i was loaded another 20 items and print the items count on console(Loaded40)
3. Again add another 20 items when scroll reaches end(see data item - 39) and so on.

In the process, I was identify the scroll view reaches it end by it max scroll extent property but sometime its value is infinity on mouse or trackpad fast scroll. So i did not identify the scroll view reaches its end.

In the above attached video, circle progress indicator not shown and data's are not added when we reach data item - 39 because (_controller.offset >= _controller.position.maxScrollExtent) return false due to the maxScrollExtent value(infinity)

**Comment by darshankawar** [comment]
Keeping this issue open for further analysis.

**Comment by yjbanov** [comment]
Is this reproducible on desktop (non-web)?

**Comment by Sasikumar3595** [comment]
@yjbanov,

Yes, this issue reproduced on macOS.

**Comment by yjbanov** [comment]
Removed the web label. This seems to affect all platforms and needs to be fixed in the framework.

/cc @Piinks

**Comment by Piinks** [comment]
Looks similar to use case discussed in https://github.com/flutter/flutter/issues/74404
I think this is expected. The `SliverChildBuilderDelegate` does not have a `childCount` set, so it does not know the number of items in the list or the possible extent until you reach the end and your builder returns null.

**Comment by Sasikumar3595** [comment]
@Piinks,

The above mentioned issue was replicated on some fast scrolls, other than that it is working fine and also it works correctly on mobile platform(android).

**Comment by Sasikumar3595** [comment]
@Piinks,

The above mentioned issue does not occurred all the scrolls, it occurs only some of the fast scroll. Please refer the above added video. Issue was occurred on web, desktop platform(mac and windows), it was not occurred on mobile platform(android).

**Comment by Sasikumar3595** [comment]
@Piinks ,

Any update on this?

**Comment by maheshmnj** [comment]
@Sasikumar3595 I verified this issue on both mobile and Web on stable 3.3 and master 3.7 and I am unable to reproduce the issue. The logs always prints a finite scrollextent

<details>
<summary>logs</summary>

```
Could not load custom device from config index 0: Expected enabled to be a boolean.
Launching lib/main.dart on iPhone 12 Pro in debug mode...
Xcode build done.                                            9.6s
Connecting to VM Service at ws://127.0.0.1:53773/-tIN3RJxY-Y=/ws
flutter: Loaded40
flutter: Loaded60
flutter: Loaded80
flutter: Loaded100
flutter: Loaded120
flutter: Loaded140
flutter: Loaded160
flutter: Loaded180
flutter: Loaded200
flutter: Loaded220
flutter: Loaded240
flutter: Loaded260
flutter: Loaded280
flutter: Loaded300
flutter: Loaded320
flutter: Loaded340
flutter: Loaded360
flutter: Loaded380

```
</details> 

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel master, 3.7.0-13.0.pre.131, on macOS 13.1 22C65 darwin-arm64, locale en-IN)
    • Flutter version 3.7.0-13.0.pre.131 on channel master at /Users/mahesh/Development/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 0196e6050b (2 days ago), 2022-12-31 11:10:23 -0500
    • Engine revision 932591ec04
    • Dart version 3.0.0 (build 3.0.0-76.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks
      and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.74.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 13.1 22C65 darwin-arm64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 108.0.5359.124

[✓] HTTP Host Availability
    • All required HTTP hosts are available
    
• No issues found!
```
```
[✓] Flutter (Channel stable, 3.3.9, on macOS 12.6 21G115 darwin-arm, locale en-IN)
    • Flutter version 3.3.9 on channel stable at /Users/mahesh/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b8f7f1f986 (24 hours ago), 2022-11-23 06:43:51 +0900
    • Engine revision 8f2221fbef
    • Dart version 2.18.5
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.53.20221101

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

I verified this issue with a trackpad though, Is this still reproducible on your end?

**Comment by Sasikumar3595** [comment]
@maheshmnj 

The above mentioned issue does not occurred all the scrolls, it occurs only some of the fast scroll. Issue was occurred on web, desktop platform(mac and windows), it was not occurred on mobile platform(android).

**Note**: Issue occurred on mouse scrolls on web and desktop platform(mac and windows).

**Comment by dam-ease** [comment]
I'm unable to reproduce this on my end as well on the latest `master` and `stable` channels. Tested on Web and macOS and a finite `scrollextent` is always printed out.

> The above mentioned issue does not occurred all the scrolls, it occurs only some of the fast scroll

From your last comment it seems this doesn't reproduce at all times, are there any patterns you could share that reproduces this if the issue still persists on your end?

<details><summary>stable, master flutter doctor -v</summary>
<p>

```
[!] Flutter (Channel stable, 3.19.1, on macOS 14.2.1 23C71 darwin-arm64, locale
    en-NG)
    • Flutter version 3.19.1 on channel stable at
      /Users/damilolaalimi/sdks/flutter
    ! Warning: `dart` on your path resolves to
      /opt/homebrew/Cellar/dart/3.1.5/libexec/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/damilolaalimi/sdks/flutter.
      Consider adding /Users/damilolaalimi/sdks/flutter/bin to the front of your
      path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision abb292a07e (8 days ago), 2024-02-20 14:35:05 -0800
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/damilolaalimi/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[!] Android Studio (version unknown)
    • Android Studio at /Users/damilolaalimi/Downloads/Android Studio
      Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build
      17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.86.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.78.0

[✓] VS Code (version 1.83.1)
    • VS Code at /Users/damilolaalimi/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.78.0

[✓] Connected device (4 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554             • android-arm64  •
      Android 14 (API 34) (emulator)
    • Damilola’s iPhone (mobile)  • 00008110-001964480AE1801E • ios            •
      iOS 17.1.1 21B91
    • macOS (desktop)             • macos                     • darwin-arm64   •
      macOS 14.2.1 23C71 darwin-arm64
    • Chrome (web)                • chrome                    • web-javascript •
      Google Chrome 122.0.6261.69

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
``` 
```
[!] Flutter (Channel master, 3.20.0-15.0.pre.26, on macOS 14.2.1 23C71 darwin-arm64, locale en-NG)
    • Flutter version 3.20.0-15.0.pre.26 on channel master at /Users/damilolaalimi/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/3.1.5/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/damilolaalimi/fvm/versions/master. Consider adding /Users/damilolaalimi/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 82ffde7ddf (3 hours ago), 2024-02-28 07:39:21 +0000
    • Engine revision 8acc96d405
    • Dart version 3.4.0 (build 3.4.0-185.0.dev)
    • DevTools version 2.33.0-dev.11
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/damilolaalimi/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[!] Android Studio (version unknown)
    • Android Studio at /Users/damilolaalimi/Downloads/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.86.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.78.0

[✓] VS Code (version 1.83.1)
    • VS Code at /Users/damilolaalimi/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.78.0

[✓] Connected device (5 available)
    • sdk gphone64 arm64 (mobile)     • emulator-5554             • android-arm64  • Android 14 (API 34) (emulator)
    • Damilola’s iPhone (mobile)      • 00008110-001964480AE1801E • ios            • iOS 17.1.1 21B91
    • macOS (desktop)                 • macos                     • darwin-arm64   • macOS 14.2.1 23C71 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad     • darwin         • macOS 14.2.1 23C71 darwin-arm64
    • Chrome (web)                    • chrome                    • web-javascript • Google Chrome 122.0.6261.69

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
``` 

</p>
</details>
